powerscale/helper/quota_helper.go

```diff
@@ -176,3 +178,34 @@ func IsQuotaParamInvalid(plan models.QuotaResource) error {
 	}
 	return nil
 }
+
+// UpdateThresholds compares given prior and updated states
+// Due to unknown reason, plugin framework would unmarshal decimal as weird float
+// e.g. advisory_percentage = 10.4 in tf state and advisory_percentage as NumberAttribute in schema
+// would be unmarshalled as something different from types.NumberValue(big.NewFloat())
+// This could cause the problem of
+// Currently use this function to return the prior thresholds if both are semantically equal
+func UpdateThresholds(prior, updated types.Object) types.Object {
+	// no comparison need in this case
+	if prior.IsNull() || updated.IsNull() {
+		return updated
+	}
+	priorAttrs := prior.Attributes()
+	for key, priorVal := range priorAttrs {
+		updatedVal, exists := updated.Attributes()[key]
+		if !exists || priorVal.IsNull() || updatedVal.IsNull() {
+			return updated
+		}
+		if numVal1, ok := priorVal.(basetypes.NumberValue); ok {
+
+			if numVal2, ok := updatedVal.(basetypes.NumberValue); ok {
+				if numVal1.ValueBigFloat().String() != numVal2.ValueBigFloat().String() {
```

Comment:
Is string comparison preferable to `numVal1.ValueBigFloat().Cmp(numVal2.ValueBigFloat())` ?

Reply:
Actually, that is the problem. 
For example, although the value unmarshalled from `.tfstate` & `types.NumberValue(big.NewFloat(10.4))` are 10.4
![image](https://github.com/user-attachments/assets/5de6e62d-6ef0-4a2b-b67e-d505c1c2f199)
But they are actually different `BigFloat ` and` .Cmp()` would return false:
![image](https://github.com/user-attachments/assets/f326b5e2-6bf7-472c-b19a-42937b94dec7)
![image](https://github.com/user-attachments/assets/309bcf64-d5c8-4cc4-a993-ed4110a548a2)

Due to this reason, `terraform plan` would detect the difference even if `.tf` file is not changed.

Reply:
Thanks for clarifying. It is interesting the two numbers look the same...